ballerina/googlesheets_client.bal

```diff
@@ -0,0 +1,520 @@
+                    }
+                } else {
+                    return <error>error(string `ColumnId for the field : ${primaryKey} cannot be found.`);
+                }
+            }
+        } else {
+            string? columnId = self.fieldMetadata.get(self.keyFields[0])["columnId"];
+            if (columnId !is ()) {
+                string dataType = typeMap.get(self.keyFields[0]).toString();
+                if (dataType == "string") {
+                    whereClause = string `${columnId} = '${key.toString()}'`;
+                } else {
+                    whereClause = string `${columnId} = ${key.toString()}`;
+                }
+
+            } else {
+                return <error>error(string `ColumnId for the field : ${self.keyFields[0]} cannot be found.`);
+            }
+        }
+        return whereClause;
+
+    }
+
+    private isolated function generateColumnIds(string[] fields) returns string|error {
+        string columnIds = "";
+        foreach string fieldMetadataKey in fields {
+            if self.fieldMetadata.hasKey(fieldMetadataKey) == false {
+                return <error>error("Error: no such a key:" + fieldMetadataKey);
+            }
+            SheetFieldMetadata fieldMetadata = self.fieldMetadata.get(fieldMetadataKey);
+            if (columnIds != "") {
+                columnIds += ", ";
+            }
+            columnIds += fieldMetadata.columnId;
+        }
+        return columnIds;
+    }
+
+    public isolated function getKey(anydata|record {} 'object) returns anydata|record {} {
+        record {} keyRecord = {};
+
+        if self.keyFields.length() == 1 && 'object is record {} {
+            return 'object[self.keyFields[0]];
+        }
+
+        if 'object is record {} {
+            foreach string key in self.keyFields {
+                keyRecord[key] = 'object[key];
+            }
+        } else {
+            keyRecord[self.keyFields[0]] = 'object;
+        }
+        return keyRecord;
+    }
+
+    private isolated function removeUnwantedFields(record {} 'object, string[] fields) {
+        foreach string keyField in self.keyFields {
+            if fields.indexOf(keyField) is () {
+                _ = 'object.remove(keyField);
+            }
+        }
+    }
+
+    private isolated function timeToString((time:Date|time:TimeOfDay|time:Civil|time:Utc) timeValue) returns string|error {
+        if timeValue is time:TimeOfDay {
+            return string `${timeValue.hour}:${timeValue.minute}:${(timeValue.second).toString()}`;
+        } else if timeValue is time:Date {
+            return string `${timeValue.day}:${timeValue.month}:${timeValue.year}`;
+        } else if timeValue is time:Civil {
+            return time:civilToString(timeValue);
+        } else if timeValue is time:Utc {
+            return time:utcToString(timeValue);
+        } else {
+            return <Error>error("Error: unsupported time format");
+        }
```

Comment:
```suggestion
        if timeValue is time:TimeOfDay {
            return string `${timeValue.hour}:${timeValue.minute}:${(timeValue.second).toString()}`;
        }
        
        if timeValue is time:Date {
            return string `${timeValue.day}:${timeValue.month}:${timeValue.year}`;
        }
        
        if timeValue is time:Civil {
            return time:civilToString(timeValue);
        }
        
        return time:utcToString(timeValue);
```

Reply:
updated in [4346c39](https://github.com/ballerina-platform/module-ballerina-persist/pull/201/commits/4346c3959a465aeb05de0872b94ea1056a715a9a)